framework/src/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -1195,6 +1195,10 @@ public AnnotatedTypeMirror getAnnotatedTypeLhs(Tree lhsTree) {
             case MEMBER_SELECT:
             case ARRAY_ACCESS:
                 res = getAnnotatedType(lhsTree);
+                if (dependentTypesHelper != null && lhsTree.getKind() == Kind.VARIABLE) {
```

Comment:
Also, in general, if you think you found a framework bug, it's good to confirm it using more than one checker.